flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala

```diff
@@ -554,10 +680,14 @@ abstract class TableEnvironment(val config: TableConfig) {
     * @param name The table name to check.
     * @return true, if a table is registered under the name, false otherwise.
     */
-  protected def isRegistered(name: String): Boolean = {
+  protected[flink] def isRegistered(name: String): Boolean = {
     rootSchema.getTableNames.contains(name)
   }
 
+  protected[flink] def getTable(name: String): org.apache.calcite.schema.Table = {
```

Comment:
can we make this a `private` method? Otherwise it will be visible from Java and confuse users (the returned Table is not a Table API Table but a Calcite Table).